In [1]:
import pandas as pd
import numpy as np
from src import soporte_limpieza as sl
pd.set_option("display.max_columns",None)
pd.set_option('display.max_colwidth', None) # No cortar el contenido de las celdas
pd.options.display.float_format = '{:.2f}'.format

>Carga del archivo y análisis visual de las columnas: Esto nos dará un mayo entendimiento del contexto de las columnas  

In [2]:
df = pd.read_pickle("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-unidos.pkl")
df.sample()

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
30097   Ministério da Defesa          52211   

                                           NOMBRE ÓRGANO  \
30097  Caixa de Financiamento Imobiliário da Aeronáutica   

       CÓDIGO UNIDAD GESTORA                        NOMBRE UNIDAD GESTORA  \
30097                 123001  CAIXA FINANCIAMENTO IMOBILIARIO AERONAUTICA   

      CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
30097  Receitas Correntes  Outras Receitas Correntes   

                                        TIPO INGRESO  \
30097  Multas administrativas, contratuais e judicia   

                                            DETALLE  \
30097  MULTAS E JUROS PREVISTOS EM CONTRATOS-PRINC.   

       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  VALOR REALIZADO  \
30097                        0.00           0.00            14.14   

       PORCENTAJE REALIZADO FECHA LANZAMIENTO  AÑO EJERCICIO  
30097                  0.00        2017-01-02           2017

> Ver el tamaño de una estructura de datos. Esto es de gran utilidad para verificar cuántos datos se cargaron desde un archivo.

In [3]:
df.shape

(1026299, 15)

> Utilizamos el atributo df.columns en pandas para conocer el nombre de todas las columnas de una manera rápida

In [4]:
df.columns

Index(['NOMBRE ÓRGANO SUPERIOR', 'CÓDIGO ÓRGANO', 'NOMBRE ÓRGANO',
       'CÓDIGO UNIDAD GESTORA', 'NOMBRE UNIDAD GESTORA', 'CATEGORÍA ECONÓMICA',
       'ORIGEN INGRESO', 'TIPO INGRESO', 'DETALLE',
       'VALOR PREVISTO ACTUALIZADO', 'VALOR LANZADO', 'VALOR REALIZADO',
       'PORCENTAJE REALIZADO', 'FECHA LANZAMIENTO', 'AÑO EJERCICIO'],
      dtype='object')

> Obtenemos toda la información del DataFrame para analizarlo

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1026299 entries, 0 to 134592
Data columns (total 15 columns):
 #   Column                      Non-Null Count    Dtype         
---  ------                      --------------    -----         
 0   NOMBRE ÓRGANO SUPERIOR      1026299 non-null  object        
 1   CÓDIGO ÓRGANO               1026299 non-null  int64         
 2   NOMBRE ÓRGANO               1026299 non-null  object        
 3   CÓDIGO UNIDAD GESTORA       1026299 non-null  int64         
 4   NOMBRE UNIDAD GESTORA       1026299 non-null  object        
 5   CATEGORÍA ECONÓMICA         1026299 non-null  object        
 6   ORIGEN INGRESO              1026299 non-null  object        
 7   TIPO INGRESO                1026299 non-null  object        
 8   DETALLE                     1026299 non-null  object        
 9   VALOR PREVISTO ACTUALIZADO  1026299 non-null  float64       
 10  VALOR LANZADO               1026299 non-null  float64       
 11  VALOR REALIZADO             10

**Insights:** 

- El DataFrame contiene 1026299 registros y 15 columnas con información sobre la ejecución de ingresos públicos en Brasil durante los años 2013 al 2021.

- Todas las columnas tienen datos completos después de trabajar sobre los datos nulos que nos dio la columna "FECHA DE LANZAMIENTO".

- Los tipos de datos incluyen 2 columnas numéricas enteras (int64), 7 categóricas (object), 4 numérica flotante (float64) y 2 de fecha (datetime64[ns] e int32). 

- El DataFrame ocupa aproximadamente 121.4 MB en memoria.

> Analisis descriptivo: Análisis estadístico básico, organizado en filas para facilitar la lectura, comparación y diagnóstico inicial de las columnas de un DataFrame.

In [6]:
df_descriptivos_numericos = df.describe().T
df_descriptivos_numericos

count                           mean  \
CÓDIGO ÓRGANO              1026299.00                       31435.16   
CÓDIGO UNIDAD GESTORA      1026299.00                      231293.74   
VALOR PREVISTO ACTUALIZADO 1026299.00                    27611421.05   
VALOR LANZADO              1026299.00                     5242806.11   
VALOR REALIZADO            1026299.00                    24980422.31   
PORCENTAJE REALIZADO       1026299.00                          80.31   
FECHA LANZAMIENTO             1026299  2018-09-20 14:26:49.968756480   
AÑO EJERCICIO              1026299.00                        2018.23   

                                            min                  25%  \
CÓDIGO ÓRGANO                          20101.00             25201.00   
CÓDIGO UNIDAD GESTORA                 110005.00            154003.00   
VALOR PREVISTO ACTUALIZADO        -214773743.00                 0.00   
VALOR LANZADO                  -148347267246.91                 0.00   
VALOR REALIZADO                -156285934188.03               260.90   
PORCENTAJE REALIZADO                 -114552.00                 0.00   
FECHA LANZAMIENTO           2013-12-31 00:00:00  2017-03-31 00:00:00   
AÑO EJERCICIO                           2013.00              2017.00   

                                            50%                  75%  \
CÓDIGO ÓRGANO                          26298.00             36000.00   
CÓDIGO UNIDAD GESTORA                 170013.00            250088.00   
VALOR PREVISTO ACTUALIZADO                 0.00                 0.00   
VALOR LANZADO                              0.00                 0.00   
VALOR REALIZADO                         3022.32             40853.85   
PORCENTAJE REALIZADO                       0.00                 0.00   
FECHA LANZAMIENTO           2018-08-27 00:00:00  2020-02-12 00:00:00   
AÑO EJERCICIO                           2018.00              2020.00   

                                            max           std  
CÓDIGO ÓRGANO                          91214.00      10382.65  
CÓDIGO UNIDAD GESTORA                 913001.00     144178.42  
VALOR PREVISTO ACTUALIZADO     1603521711208.00 3393367175.13  
VALOR LANZADO                   357160677863.97  810883695.36  
VALOR REALIZADO                 771117711060.95 1406892548.84  
PORCENTAJE REALIZADO                72363772.00      71481.11  
FECHA LANZAMIENTO           2021-12-03 00:00:00           NaN  
AÑO EJERCICIO                           2021.00          1.74

**Insights:** 


1.	CÓDIGO ÓRGANO: Código numérico que identifica la entidad gubernamental superior.

    - Al tratarse de una columna con códigos que identifican la entidad gubernamental superior, no podemos sacar conclusiones relevantes del análisis descriptivo.

2.	CÓDIGO UNIDAD GESTORA: Código numérico de la unidad gestora responsable.

	- De igual manera que en el caso anterior, no podemos no podemos sacar conclusiones relevantes del análisis descriptivo.

3.	VALOR PREVISTO ACTUALIZADO: Monto actualizado del ingreso previsto (formato texto).

	- Podemos observar que los percentiles son 0, lo que quiere decir que la mayoría de los datos son 0. Lo cual quiere decir que el valor previsto actualizado en la mayoría de los casos es 0.
	- El valor mínimo es negativo y con una cantidad bastante grande (-214.773.743), algo que parece extraño al ser un "valor previsto actualizado"
	- El valor máximo (1.603.521.711.208) es muy alto si lo comparamos con la media (27.611.321). Además, si lo comparamos con la desviación estándar (3.393.367.175), vemos que hay una gran dispersión en los datos. Es decir, hay datos muy altos y muy bajos, por lo que el valor máximo es algo puntual.

4.	VALOR LANZADO: Monto que fue registrado como lanzado

	- Vemos que nos sucede algo similar al anterior caso. Los percentiles son 0, lo cuál quiere decir que la mayoría de los datos son 0.
	- El valor mínimo (-148.347.267.246) es negativo, lo cuál es extraño al ser un valor lanzado. 
	- Si comparamos la media (5.242.806) con el valor máximo (357.160.677.863) hay mucha diferencia y, viendo la desviación estándar (810.883.695) podemos confirmar que hay uan gran desviación de los datos. Por lo que hay mucha diferencia entre unos valores y otros.

5.	VALOR REALIZADO: Monto realmente recaudado

	- El 25% de los datos están por encima de 260,905, el 50% (3.022,32) refleja que la mitad de los registros tienen ese valor o menos y el 75% indica que el 75% de los registros tiene valores por dejabjo de 408.53,855. Por lo que podemos afirmar que en la mayoría de los casos los registros tienden a tener valores pequeños.
	- El valor mínimo es de -156.285.934.188,029999, algo bastante extraño al tratarse de de cantidades recaudadas.
	- El valor máximo es demasiado alto en comparación con el promedio y la desviación estándar nos dice que hay mucha dispersión de los datos.

6.	PORCENTAJE REALIZADO: Porcentaje de ejecución respecto al valor previsto.

	- El principal dato a destacar es que hay datos en negativo, lo cual quiere decir que la ejecución del valor está por debajo del valor previsto.
	- El valor promedio de 80,3% quiere decir que en la mayoría de los casos el porcentaje de ejecución es bastante alto.
	- Gracias a los percentiles podemos ver que la mayoría de datos son 0.
	- La desviación estándar nos dice que hay mucha dispersión de datos, lo cuál quiere decir que hay porcentajes muy altos y muy bajos.

7.	FECHA LANZAMIENTO: Fecha en la que se registró la ejecución del ingreso.

	- El valor mínimo (2013-12-31) lo cuál quiere decir que la primera fecha de la base de datos es del 31 de diciembre de 2013 y la última del 3 de diciembre de 2021.

8.	AÑO EJERCICIO: Año correspondiente a la ejecución de los ingresos.

	- 

Conclusión:


Los datos muestran una alta variabilidad, con valores tanto negativos como cercanos a cero, lo cuál es llamativo al tratarse de ingresos, es decir, los ingresos registrados tienen una gran dispersión en relación al valor previsto y lanzado. Por lo que hay que realizar un análisis de los datos para saber qué está pasando.
La ejecución promedio respecto al valor previsto es alta (80%), aunque muchos registros tienen un porcentaje de ejecución de 0%, lo que refleja que una parte significativa de los ingresos aún no se ha realizado o ejecutado.

In [7]:
df_descriptivos_object = df.describe(include = "O").T
df_descriptivos_object

count unique  \
NOMBRE ÓRGANO SUPERIOR  1026299     25   
NOMBRE ÓRGANO           1026299    287   
NOMBRE UNIDAD GESTORA   1026299    356   
CATEGORÍA ECONÓMICA     1026299      5   
ORIGEN INGRESO          1026299     15   
TIPO INGRESO            1026299     63   
DETALLE                 1026299   1886   

                                                                         top  \
NOMBRE ÓRGANO SUPERIOR                                Ministério da Educação   
NOMBRE ÓRGANO           Ministério da Economia - Unidades com vínculo direto   
NOMBRE UNIDAD GESTORA                SETORIAL ORCAMENTARIA E FINANCEIRA / ME   
CATEGORÍA ECONÓMICA                                       Receitas Correntes   
ORIGEN INGRESO                                     Outras Receitas Correntes   
TIPO INGRESO                    Serviços Administrativos e Comerciais Gerais   
DETALLE                          SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC.   

                          freq  
NOMBRE ÓRGANO SUPERIOR  361891  
NOMBRE ÓRGANO           119811  
NOMBRE UNIDAD GESTORA   108481  
CATEGORÍA ECONÓMICA     979596  
ORIGEN INGRESO          322907  
TIPO INGRESO            278157  
DETALLE                 159039

**Insights:** 
Análisis conciso de los datos:

1. NOMBRE ÓRGANO SUPERIOR: Nombre de la entidad gubernamental superior.

	- De todos los registros, hay 25 órganos superiores diferentes, siendo el Ministerio de Eucación el más frecuente.

2.	NOMBRE ÓRGANO: Nombre de la entidad gubernamental específica.

	- El ministerio de Economía es el órgano más frecuente de los 287 que hay en la base de datos.

3.	NOMBRE UNIDAD GESTORA: Nombre de la unidad gestora.

	- La unidad gestora Setorial Orcamentaria y financiera es la más freciente dentro de los 356 registros únicos que hay.

4.	CATEGORÍA ECONÓMICA: Clasificación económica de los ingresos (por ejemplo, "Receitas Correntes").

	- Hay 5 categorías económicas y Recetas Corrientes es la más frecuente.

5.	ORIGEN INGRESO: Fuente específica del ingreso (por ejemplo, "Outras Receitas Correntes").
	
	- Hay 15 tipos de fuentes específicas de los ingresos y ''Otras recetas corrientes'' es la más frecuente.

6.	TIPO INGRESO: Tipo de ingreso dentro de la fuente (por ejemplo, "Demais receitas correntes").
	
	- El tipo de ingreso tiene 63 diferentes en toda la base de datos y los Serviços Administrativos e Comerciais Gerais son los más frecuentes.

7.	DETALLE: Detalle adicional del tipo de ingreso (por ejemplo, "Receita de honorários de advogados").

	- Es la variable categórica con más valore súnicos y SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC. es el detalle más repetido.

Conclusión:

Estos datos reflejan las características cualitiativas de los ingresos públicos de Brasil. Destacando las receitas correntes como la categoría económica más recurrente y otras recetas correntes como el principal origen de los ingresos.

> Analisis de los valores unicos de las variables categoricas 

In [8]:
sl.analisis_categoria(df,"NOMBRE ÓRGANO SUPERIOR")

NOMBRE ÓRGANO SUPERIOR   count  porcentaje
0                          Ministério da Educação  361891       35.26
1                          Ministério da Economia  207574       20.23
2                            Ministério da Defesa   95490        9.30
3   Ministério da Agricultura, Pecuária e Abastec   46183        4.50
4       Ministério da Justiça e Segurança Pública   39980        3.90
5                    Ministério da Infraestrutura   37476        3.65
6                             Ministério da Saúde   33380        3.25
7                   Ministério de Minas e Energia   33014        3.22
8   Ministério da Ciência, Tecnologia, Inovações    32846        3.20
9          Ministério do Desenvolvimento Regional   32725        3.19
10                    Ministério das Comunicações   29485        2.87
11                    Ministério do Meio Ambiente   27151        2.65
12                          Ministério do Turismo   27096        2.64
13                        Ministério da Cidadania    5898        0.57
14                       Presidência da República    4976        0.48
15                       Advocacia-Geral da União    3899        0.38
16             Ministério das Relações Exteriores    3563        0.35
17  Ministério da Mulher, Família e Direitos Huma    1157        0.11
18               Ministério do Trabalho e Emprego     685        0.07
19          Ministério do Desenvolvimento Agrário     666        0.06
20              Ministério da Pesca e Aquicultura     373        0.04
21  Ministério das Mulheres, Igualdade Racial, da     285        0.03
22                   Controladoria-Geral da União     252        0.02
23                          Ministério do Esporte     147        0.01
24               Ministério da Previdência Social     107        0.01

**Insights:** 

Los órganos que mayor protagonismo tienen son los de Educación y Economía con un 35,26% y un 20,23% respectivamente. El tercer órgano con más protagonismo es el Ministerio de Defensa con un 9,30%.

In [9]:
sl.analisis_categoria(df,"NOMBRE ÓRGANO")

NOMBRE ÓRGANO  \
0                        Ministério da Economia - Unidades com vínculo direto   
1                                                           Fundo do Exército   
2                                            Comando da Marinha - Fundo Naval   
3                                                   Fundo Nacional de Cultura   
4    Instituto Brasileiro do Meio Ambiente e dos Recursos Naturais Renováveis   
..                                                                        ...   
282                                    Agência de Desenvolvimento da Amazônia   
283                                             Hospital Cristo Redentor S.A.   
284                              Departamento Nacional de Estradas de Rodagem   
285                                                      Hospital Fêmina S.A.   
286                                 Companhia de Desenvolvimento de Barcarena   

      count  porcentaje  
0    119811       11.67  
1     25559        2.49  
2     23734        2.31  
3     15955        1.55  
4     13745        1.34  
..      ...         ...  
282       2        0.00  
283       2        0.00  
284       1        0.00  
285       1        0.00  
286       1        0.00  

[287 rows x 3 columns]

**Insights:** 

Vemos que hay una gran diferencia entre el primero (11,67%) y el segundo (2,49%). El órgano con más protagonismo es el Ministério da Economia - Unidades com vínculo direto. El resto de órganos tienen máyor o menor protagonismo pero mucho más ajustado que este.

In [10]:
sl.analisis_categoria(df,"NOMBRE UNIDAD GESTORA")

NOMBRE UNIDAD GESTORA   count  porcentaje
0          SETORIAL ORCAMENTARIA E FINANCEIRA / ME  108481       10.57
1                                FUNDO DO EXERCITO   25469        2.48
2     DIRETORIA DE FINANCAS-SISTEMA PARA O PAIS/MM   23532        2.29
3     IBAMA - INST.BRASILEIRO MEIO AMBIENTE/MATRIZ   13698        1.33
4                                 EMBRAPA-SETORIAL   13641        1.33
..                                             ...     ...         ...
351                   HOSPITAL CRISTO REDENTOR S/A       2        0.00
352                            HOSPITAL FEMINA S/A       1        0.00
353      COMPANHIA DE DESENVOLVIMENTO DE BARCARENA       1        0.00
354  COORDENACAO DE ORCAMENTO E FINANCAS - COF/DPF       1        0.00
355  DEPARTAMENTO NAC. ESTRADAS E RODAGEM-EXTINCAO       1        0.00

[356 rows x 3 columns]

**Insights**

Vemos que sucede parecido al anterior caso. Sería interesante hacer un análisis comparativo entre los dos organismos para ver su correlación. En este caso el SETORIAL ORCAMENTARIA E FINANCEIRA / ME tiene un 10,57% de protagonismo frente a los demás que parten de un 2,48%.

Vemos que son datos muy similares a los anteriores, lo cuál puede decir que hay mucha correlación entre ambas columnas

In [11]:
sl.analisis_categoria(df,"CATEGORÍA ECONÓMICA")

CATEGORÍA ECONÓMICA   count  porcentaje
0                         Receitas Correntes  979596       95.45
1                        Receitas de Capital   30116        2.93
2   Receitas Correntes - intra-orçamentárias   16235        1.58
3                             Sem informação     257        0.03
4  Receitas de Capital - intra-orçamentárias      95        0.01

**Insights**

Las recetas correntes son las que más predominan con un 95% frente al resto de categorías. 

In [12]:
sl.analisis_categoria(df,"ORIGEN INGRESO")

ORIGEN INGRESO   count  porcentaje
0                     Outras Receitas Correntes  322907       31.46
1                           Receita de Serviços  322319       31.41
2                           Receita Patrimonial  163171       15.90
3   Impostos, Taxas e Contribuições de Melhoria   78340        7.63
4                                 Contribuições   66781        6.51
5                            Receita Industrial   18857        1.84
6                          Receita Agropecuária   13642        1.33
7                             Alienação de Bens   12418        1.21
8                   Amortizações de Empréstimos   11318        1.10
9                      Transferências Correntes    6385        0.62
10                         Operações de Crédito    5044        0.49
11           Receitas Correntes - a classificar    3429        0.33
12                   Outras Receitas de Capital    1060        0.10
13                    Transferências de Capital     371        0.04
14                               Sem informação     257        0.03

**Insights**

Vemos que el origen de los ingresos está repartido muy equitativamente entre Outras Receitas Correntes y Receita de Serviços que tienen un porcentaje casi similar, 31,46% y 31,41% respectivamente. 

In [13]:
sl.analisis_categoria(df,"TIPO INGRESO")

TIPO INGRESO   count  porcentaje
0    Serviços Administrativos e Comerciais Gerais  278157       27.10
1     Indenizações, restituições e ressarcimentos  143577       13.99
2   Multas administrativas, contratuais e judicia  115276       11.23
3   Exploração do patrimônio imobiliário do Estad   88862        8.66
4                                           Taxas   62336        6.07
..                                            ...     ...         ...
58  Receita dívida ativa alienação de estoques de       3        0.00
59  Receitas decorrentes de compensações ao Regim       2        0.00
60                     Demais receitas de capital       2        0.00
61                        Receita da dívida ativa       1        0.00
62               Integralização do capital social       1        0.00

[63 rows x 3 columns]

**Insights**

los tipos de ingresos más predominantes son: Serviços Administrativos e Comerciais Gerais (27,10%), Indenizações, restituições e ressarcimentos (13,99%) y Multas administrativas, contratuais e judicia (11,23%)

In [14]:
sl.analisis_categoria(df,"DETALLE")

DETALLE   count  porcentaje
0       SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC.  159039       15.50
1     INSCR.EM CONCURSOS E PROC.SELETIVOS-PRINCIPAL   74638        7.27
2                ALUGUEIS E ARRENDAMENTOS-PRINCIPAL   71040        6.92
3      MULTAS PREVISTAS EM LEGISL.ESPECIFICA-PRINC.   44580        4.34
4     RESTIT.DE DESPESAS DE EXERC.ANTERIORES-PRINC.   31513        3.07
...                                             ...     ...         ...
1881  CONTRIBUIÇÃO PARA O PIS/PASEP-JUROS-DÍV.ATIVA       1        0.00
1882  RESTITUICAO DE RECURSOS DE FOMENTO-DIV. ATIVA       1        0.00
1883       ROYALTIES MINIMOS PELA PRODOCAO EM TERRA       1        0.00
1884        RECEITA INDUSTRIAL-MULTAS E JUROS-INTRA       1        0.00
1885    ITR-MUNICIPIOS CONVENIADOS-DIV.AT.-MUL.JUR.       1        0.00

[1886 rows x 3 columns]

**Insights**

El detalle predominante es SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC. con un 15,50% frente al segundo que es INSCR.EM CONCURSOS E PROC.SELETIVOS-PRINCIPAL con un 7,27%

> Distribución de ingresos por categoría económica

In [15]:
df.head(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
0  Advocacia-Geral da União          63000   

                                            NOMBRE ÓRGANO  \
0  Advocacia-Geral da União - Unidades com vínculo direto   

   CÓDIGO UNIDAD GESTORA                          NOMBRE UNIDAD GESTORA  \
0                 110060  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

  CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
0  Receitas Correntes  Outras Receitas Correntes   

                                    TIPO INGRESO  \
0  Bens, Direitos e Valores Incorporados ao Patr   

                                         DETALLE  VALOR PREVISTO ACTUALIZADO  \
0  REC.DIVIDA ATIVA NAO TRIBUTARIA DE OUTRAS REC                        0.00   

   VALOR LANZADO  VALOR REALIZADO  PORCENTAJE REALIZADO FECHA LANZAMIENTO  \
0           0.00          1297.13                  0.00        2013-12-31   

   AÑO EJERCICIO  
0           2013

- Análisis de las categorías de ingresos más significativas y su participación en los ingresos totales.

In [16]:
categorias_ingresos = df.groupby("CATEGORÍA ECONÓMICA")[["VALOR PREVISTO ACTUALIZADO","VALOR LANZADO","VALOR REALIZADO"]].sum()
categorias_ingresos["DIFERENCIA"] = categorias_ingresos["VALOR PREVISTO ACTUALIZADO"] - categorias_ingresos["VALOR REALIZADO"]
categorias_ingresos = categorias_ingresos.reset_index()
categorias_ingresos.sort_values(by="DIFERENCIA", ascending=False)

CATEGORÍA ECONÓMICA  VALOR PREVISTO ACTUALIZADO  \
2                        Receitas de Capital           14485753431206.59   
0                         Receitas Correntes           13549526943206.14   
4                             Sem informação                        0.00   
1   Receitas Correntes - intra-orçamentárias             291485025919.00   
3  Receitas de Capital - intra-orçamentárias              10808411537.00   

     VALOR LANZADO   VALOR REALIZADO       DIFERENCIA  
2       3813949.42 12407136680065.79 2078616751140.80  
0 5377870503243.72 12909343128909.43  640183814296.71  
4    2812347576.39             46.57           -46.57  
1             0.00   292651790699.89   -1166764780.89  
3             0.00    28250835724.33  -17442424187.33

Vemos que en todos los casos se ha superado el valor lanzado, excepto en aquellas categorías sin información, que se estimó una cantidad que no se ha alcanzado por muchísima diferencia.

En el caso de las Receitas Correntes - intra-orçamentárias y de las Receitas de Capital - intra-orçamentárias  el valor realizado ha superado incluso al valor previsto actualizado, en este segundo caso pasando de 10,808,411,537.00 a 28,250,835,724.33. 

In [17]:
origen_ingresos = df.groupby("ORIGEN INGRESO")[["VALOR PREVISTO ACTUALIZADO","VALOR LANZADO","VALOR REALIZADO"]].sum()
origen_ingresos = origen_ingresos.reset_index()

origen_ingresos["DIFERENCIA"] = origen_ingresos["VALOR PREVISTO ACTUALIZADO"] - origen_ingresos["VALOR REALIZADO"]
origen_ingresos.sort_values(by="DIFERENCIA", ascending=False)

ORIGEN INGRESO  VALOR PREVISTO ACTUALIZADO  \
4                          Operações de Crédito           12787671290356.19   
2                                 Contribuições            7415248660408.00   
3   Impostos, Taxas e Contribuições de Melhoria            4510210765178.00   
0                             Alienação de Bens              69459904322.00   
5                     Outras Receitas Correntes             527337121145.00   
10                          Receita de Serviços             440758536057.11   
11           Receitas Correntes - a classificar                        0.00   
13                     Transferências Correntes               4761797513.00   
8                            Receita Industrial              20116477675.00   
14                    Transferências de Capital               1089634563.40   
7                          Receita Agropecuária                230056327.00   
12                               Sem informação                        0.00   
9                           Receita Patrimonial             922348554822.03   
1                   Amortizações de Empréstimos             468392937016.00   
6                    Outras Receitas de Capital            1169948076486.00   

      VALOR LANZADO  VALOR REALIZADO       DIFERENCIA  
4        3813949.42 9885939455484.26 2901731834871.93  
2  3220620507720.18 7043604161886.25  371644498521.75  
3  2020808903486.62 4287292710249.35  222918054928.65  
0              0.00   27033901093.04   42426003228.96  
5   136055962948.35  495575920557.84   31761200587.16  
10             0.00  410410034253.08   30348501804.03  
11       5042321.57  -15471989712.32   15471989712.32  
13             0.00    3780118794.99     981678718.01  
8              0.00   19256167707.50     860309967.50  
14             0.00     478697459.15     610937104.25  
7              0.00     212916430.85      17139896.15  
12    2812347576.39            46.57           -46.57  
9      380086767.00  957334879441.78  -34986324619.75  
1              0.00  799034542267.89 -330641605251.89  
6              0.00 1722900919485.78 -552952842999.78

En el caso del análisis del origen de los ingresos podemos ver una conclusión similiar a la anterior. Vemos que en la mayoría de los casos se ha superado el valor lanzado. Pero hay tres orígenes que la diferencia entre el valor previsto actualizado y el valor realizado es negativo, lo cuál quiere decir que se ha superado el valor previsto.

> Análisis Temporal

In [25]:
df.head(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
0  Advocacia-Geral da União          63000   

                                            NOMBRE ÓRGANO  \
0  Advocacia-Geral da União - Unidades com vínculo direto   

   CÓDIGO UNIDAD GESTORA                          NOMBRE UNIDAD GESTORA  \
0                 110060  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

  CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
0  Receitas Correntes  Outras Receitas Correntes   

                                    TIPO INGRESO  \
0  Bens, Direitos e Valores Incorporados ao Patr   

                                         DETALLE  VALOR PREVISTO ACTUALIZADO  \
0  REC.DIVIDA ATIVA NAO TRIBUTARIA DE OUTRAS REC                        0.00   

   VALOR LANZADO  VALOR REALIZADO  PORCENTAJE REALIZADO FECHA LANZAMIENTO  \
0           0.00          1297.13                  0.00        2013-12-31   

   AÑO EJERCICIO   AÑO  MES  
0           2013  2013   12

In [32]:
tendencias_año = df.groupby("AÑO EJERCICIO")[["VALOR PREVISTO ACTUALIZADO","VALOR LANZADO","VALOR REALIZADO"]].sum()
tendencias_año = tendencias_año.reset_index()
tendencias_año["DIFERENCIA"] = tendencias_año["VALOR PREVISTO ACTUALIZADO"] - tendencias_año["VALOR REALIZADO"]
tendencias_año.sort_values(by="DIFERENCIA", ascending=False)



AÑO EJERCICIO  VALOR PREVISTO ACTUALIZADO    VALOR LANZADO  \
4           2017            3414379614681.84  459897488081.78   
5           2018            3505648331810.28  468421184578.15   
8           2021            4180716471387.52  359788769214.59   
0           2013            2174650946322.04 1142556841623.94   
6           2019            3261378698948.11  343736377894.56   
2           2015            2878827555613.92  307296621471.69   
1           2014            2392129226119.23 1266561021071.51   
3           2016            2952696248075.54  325948228390.24   
7           2020            3577146718910.25  706480132443.07   

   VALOR REALIZADO      DIFERENCIA  
4 2558047412866.96 856332201814.88  
5 2942387872873.97 563260458936.31  
8 3876372024873.04 304344446514.48  
0 1893613719099.78 281037227222.26  
6 2991564738258.26 269813960689.85  
2 2665685375159.13 213142180454.79  
1 2238551079327.88 153578146791.35  
3 2839711583462.82 112984664612.72  
7 3631448629524.17 -54301910613.92

In [33]:
df.head(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
0  Advocacia-Geral da União          63000   

                                            NOMBRE ÓRGANO  \
0  Advocacia-Geral da União - Unidades com vínculo direto   

   CÓDIGO UNIDAD GESTORA                          NOMBRE UNIDAD GESTORA  \
0                 110060  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

  CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
0  Receitas Correntes  Outras Receitas Correntes   

                                    TIPO INGRESO  \
0  Bens, Direitos e Valores Incorporados ao Patr   

                                         DETALLE  VALOR PREVISTO ACTUALIZADO  \
0  REC.DIVIDA ATIVA NAO TRIBUTARIA DE OUTRAS REC                        0.00   

   VALOR LANZADO  VALOR REALIZADO  PORCENTAJE REALIZADO FECHA LANZAMIENTO  \
0           0.00          1297.13                  0.00        2013-12-31   

   AÑO EJERCICIO   AÑO  MES  
0           2013  2013   12

In [48]:
df["MES"] = df["FECHA LANZAMIENTO"].dt.month_name()
df["MES_NUMERO"] = df["FECHA LANZAMIENTO"].dt.month
df.sample(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
143380  Ministério da Infraestrutura          39000   

                                                     NOMBRE ÓRGANO  \
143380  Ministério da Infraestrutura - Unidades com vínculo direto   

        CÓDIGO UNIDAD GESTORA                          NOMBRE UNIDAD GESTORA  \
143380                 390002  SUBSECRETARIA DE PLAN.,ORCAM.E ADMIN.SETORIAL   

       CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
143380  Receitas Correntes  Outras Receitas Correntes   

                                       TIPO INGRESO  \
143380  Indenizações, restituições e ressarcimentos   

                                DETALLE  VALOR PREVISTO ACTUALIZADO  \
143380  OUTROS RESSARCIMENTOS-PRINCIPAL                        0.00   

        VALOR LANZADO  VALOR REALIZADO  PORCENTAJE REALIZADO  \
143380           0.00            51.30                  0.00   

       FECHA LANZAMIENTO  AÑO EJERCICIO   AÑO  MES  MES_NUMERO  
143380        2016-05-11           2016  2016  May           5

In [56]:
tendencias_mes = df.groupby(["MES", "MES_NUMERO"])[["VALOR PREVISTO ACTUALIZADO","VALOR LANZADO","VALOR REALIZADO"]].sum()
tendencias_mes.reset_index()
tendencias_mes["DIFERENCIA"] = tendencias_mes["VALOR PREVISTO ACTUALIZADO"] - tendencias_mes["VALOR REALIZADO"]
tendencias_mes.sort_values(by="MES_NUMERO", ascending=True)

VALOR PREVISTO ACTUALIZADO    VALOR LANZADO  \
MES       MES_NUMERO                                                
January   1                    16711007147653.00   15076064130.74   
February  2                                 0.00  -23904264713.16   
March     3                                 0.00   10260220118.57   
April     4                     4180716471387.52     644696348.51   
May       5                                 0.00   53008836626.70   
June      6                          49838878.54 2044446443449.22   
July      7                         101407416.09  411290397497.71   
August    8                          43637746.28   92572614076.49   
September 9                                 0.00   59935127566.93   
October   10                         47580732.11  -55155034177.98   
November  11                                0.00   -1936730570.26   
December  12                    7445607728055.19 2774448294416.06   

                      VALOR REALIZADO        DIFERENCIA  
MES       MES_NUMERO                                     
January   1          1933735689252.62 14777271458400.38  
February  2          1195699037131.18 -1195699037131.18  
March     3          1666703302836.62 -1666703302836.62  
April     4          1591284527803.02  2589431943584.50  
May       5          1282092166671.36 -1282092166671.36  
June      6          1259371774462.61 -1259321935584.07  
July      7          1683137099845.77 -1683035692429.68  
August    8          1700053802235.86 -1700010164489.58  
September 9          1890279011080.10 -1890279011080.10  
October   10         1734953591607.39 -1734906010875.28  
November  11         1396620081395.41 -1396620081395.41  
December  12         8303452351124.07  -857844623068.88